src/JarbField.tsx

```diff
@@ -1,35 +1,32 @@
-
-// @flow
-
 import * as React from 'react';
 import { Field } from 'redux-form';
 
-import { getConfig } from './config';
-import type { Config } from './config';
-
+import { Config, getConfig } from './config';
 import { getFieldConstraintsFor, mostSpecificInputTypeFor } from './utils';
-
-import type { ConstraintsStore } from './constraints-reducer';
-import type { FieldConstraints, FieldType } from './models';
-
+import { ConstraintsStore } from './constraints-reducer';
+import { FieldConstraints, FieldType } from './models';
 import * as validators from './validators';
-import type { RequiredValidator, MinimumLengthValidator, MaximumLengthValidator, MinValueValidator, MaxValueValidator, PatternValidator } from './validators';
-
+import {
+  RequiredValidator,
+  MinimumLengthValidator,
+  MaximumLengthValidator,
+  MinValueValidator,
+  MaxValueValidator,
+  PatternValidator,
+} from './validators';
 import * as patterns from './regex';
 
-type JarbProps = {
-  validator: string,
-  label: string
+interface JarbProps {
+  validator: string;
+  label: string;
 }
 
-type Props = {
-  name: string,
-  component: React.ComponentType<*> | string,
-  jarb: JarbProps, 
-  validate?: Array<Function>
-};
-
-type State = { };
+interface Props {
+  name: string;
+  component: React.ComponentType | string;
```

Comment:
Volgens mij kunnen we dit stricter maken.
In `Field.d.ts` van van de `Field` component zien we dat `component` van de type `component?: ComponentType<WrappedFieldProps & P> | "input" | "select" | "textarea";` is.

Reply:
Is dit beter?

```js
interface Props<P = {}> {
  name: string;
  component: React.ComponentType<P> | 'input' | 'select' | 'textarea';
  jarb: JarbProps;
  validate?: Function[];
}
```

Reply:
Ja Props meegeven zal het niet veel beter op maken. Laten we het bij `React.ComponentType | 'input' | 'select' | 'textarea` houden.